### PR TITLE
#33 Align code format with swift-format

### LINKED ISSUE #33: Align code format using swift-format
Requirements:

- Add config file for swift-format
- create GH Action that lints the source files
- optional: setup a pre-commit hook that validates the changed source files

### PR DESCRIPTION
**Pull request by henrik-dmg**
## Context

This PR adds configuration files for standardised code formatting using Apple's `swift-format`. It also adds pre-commit hooks and a GH Actions config file

## Changes

- `Scripts/lint-swift-code` simply calls `swift-format` with the also added configuration file and the paths to the Swift source files
- `Scripts/format-swift-code` does the same, but formats the code instead of linting
- `Scripts/configure-hooks` symlinks all scripts in `Scripts/hooks` into the hooks directory of the repository
  - that way we can track the hook scripts in git and change them without having to re-configure the hooks
- `Scripts/hooks/pre-commit` runs the linting script
  - if the return code is != 0, it also prints instructions on how to fix most errors
- added new GH Actions workflow to lint Swift code on pushes and PRs to `main`

## Notes

This PR doesn't contain the actual changes that would be produced by running `Scripts/format-swift-code`. Once this PR is approved, I will create another PR that includes these changes that targets this branch before merging.

Closes #33

### CODE REVIEW

**Comment by henrik-dmg** [comment]
Note that the linting check is naturally failing because the changes produced by `swift-format` are not contained in this branch yet

File: Scripts/configure-hooks [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+HOOKS=$(git rev-parse --git-path hooks)
+ROOT=$(git rev-parse --show-toplevel)
+
+# Symlink the hooks from scripts/hooks to the $HOOKS directory
+for hook in "${ROOT}"/scripts/hooks/*; do
```
Comment (juraj-blahunka) [inline Scripts/configure-hooks]: **nitpick** The folder is `Scripts`, but we're using the lowercase version here

**Review by juraj-blahunka — APPROVED** [review]
Great job Henrik! You know that I like to nitpick and look for improvements in every file, but this looks amazing :)

File: Scripts/configure-hooks (hunk 1 above)
Comment (henrik-dmg) [reply Scripts/configure-hooks]: Ah good catch! 👀

**Review by juraj-blahunka — APPROVED** [review]

**Pull request merged**